### ISSUE TITLE
Vega expression fails with "Bitwise AND operator not supported"

### ISSUE THREAD

**Issue by sacundim** [hex-inc/vegafusion#357]
With a chart that I've been making for a long time:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: ParseError("Bitwise AND operator not supported", ErrorContext { contexts: ["Failed to parse token at position 35 in expression: (datum.lo_mid_value < datum.value) & (datum.value < datum.hi_mid_value)"] })', vegafusion-core/src/planning/stitch.rs:26:10
```

I can't quickly extract an isolated reproduction right now for this, but the issue looks straightforward enough, the Vega-Lite is using this condition in a color encoding and the Vegafusion just hasn't chosen to implement it:

```
        [...]
        "encoding": {
          "color": {
            "condition": {
              "test": "(datum.lo_mid_value < datum.value) & (datum.value < datum.hi_mid_value)",
              "value": "black"
            },
            "value": "white"
          },
         [...]
```

Workaround: As this Vega expression is just trying to do a Boolean AND, not really a bitwise AND, changing it from `&` to `&&` worked just fine.  It's a bit strange that my old chart is using that, but I'm sure that I must've copied it from some old example in Vega-Lite or Vega-Altair documentation, and therefore that I won't be the first person to encounter this.

**Comment by jonmmease** [comment]
Thanks for taking the time to report this @sacundim. I'll have to give it some thought, but if it works in Vega-Lite I definitely want to work out how to support it in VegaFusion.

It looks like DataFusion has implemented bitwise operators since I initially wrote the expression parsing and evaluation logic (https://github.com/apache/arrow-datafusion/issues/1619), so the first thing I'll try is to wire these up.

**Comment by jonmmease** [comment]
Also, please keep the reports coming! Full compatibility with the Vega specs that Vega-Lite produces is pretty much our top priority. VegaFusion needs to be something you can turn on and not worry about

**Comment by jonmmease** [comment]
Fixed in 1.4.0-rc3
